sgm/inference/api.py

```diff
@@ -242,17 +291,46 @@ def image_to_image(
             samples,
             force_uc_zero_embeddings=["txt"] if not self.specs.is_legacy else [],
             return_latents=return_latents,
-            filter=None,
+            filter=filter,
         )
 
+    def wrap_discretization(

```

Comment:
This function would be simplified with early returns.

Reply:
> This function would be simplified with early returns.

It can't, actually. Despite the name, `Txt2NoisyDiscretizationWrapper` is wrapped again around the `Img2ImgDiscretizationWrapper` when both image strength and refiner strength are used - see the original code here.

https://github.com/Stability-AI/generative-models/blob/main/scripts/demo/streamlit_helpers.py#L365C1-L376C10